### ISSUE TITLE
Artifactory Npm task - Support 'custom' commands

### ISSUE THREAD

**Issue by ashokm** [jfrog/artifactory-azure-devops-extension#276]
In the `ArtifactoryNpm@2` task in Azure DevOps, it appears that it is not possible to provide 'custom' npm commands in the `command` input.

**Note** that the `Npm@1` task in Azure DevOps **does** allow 'custom' commands to be provided.

Can the  `ArtifactoryNpm@2` task be extended to allow users to provide custom commands?

We would like to use _custom_ npm commands other than `install`, `pack and publish` and `ci`, to ensure the custom commands pull dependencies from Artifactory (and uses the `ArtifactoryNpm@2` task and its Artifactory service connection for authentication).

For example, we would like to use the `ArtifactoryNpm@2` task to be able to successfully run the following npm command:
```
npm init react-app my-app -- --template file:/path/to/template/ --use-npm
```

To be able to use the task something like this (which does not work at the moment to due the limitation):
```yaml
- task: ArtifactoryNpm@2
  displayName: 'Prepare: Create React App'
  inputs:
    artifactoryService: 'Eco-System AOL'
    sourceRepo: 'npm-virtual'
    command: 'init react-app my-app'
    arguments: '--template file=/path/to/template/ --use-npm'
```

At the moment, this results in:
```text
Starting: Prepare: Create React App
==============================================================================
Task         : Artifactory npm
Description  : Install, pack and publish npm packages from and to Artifactory while allowing to collect build-info. The collected build-info can be later published to Artifactory by the "Artifactory Publish Build Info" task.
Version      : 2.13.1
Author       : JFrog
Help         : [More Information](https://www.jfrog.com/confluence/display/JFROG/Artifactory+Azure+DevOps+Extension#ArtifactoryAzureDevOpsExtension-TriggeringNpmBuilds)
==============================================================================
Downloading: https://releases.jfrog.io/artifactory/jfrog-cli/v1/1.51.1/jfrog-cli-linux-386/jfrog
Caching tool: jfrog 1.51.1 x64
Running jfrog-cli from /opt/hostedtoolcache/jfrog/1.51.1/x64/jfrog.
JFrog CLI version: 1.51.1
Finishing: Prepare: Create React App
```

**Comment by eyalbe4** [comment]
@ashokm - 
Thanks for raising this requirement. We understand why this is needed and will promote this addition.
The Azure DevOps extension uses JFrog CLI behind the scenes, and we're already working on enhancing JFrog CLI's npm commands, to support any npm command. This is planned to be included as part of the next minor release of JFrog CLI, currently planned to be released by the end of November. The next step would be to allow the Azure DevOps extension to use the enhanced JFrog CLI command.
The timeline for this activity is not finalized yet, but we hope to be able to deliver this soon.

**Comment by RobiNino** [comment]
Hi @ashokm ,
As part of our effort of enhancing the integration between the JFrog Platform and Azure DevOps, we recently released a new extension to Azure DevOps - [JFrog Azure DevOps Extension](https://marketplace.visualstudio.com/items?itemName=JFrog.jfrog-azure-devops-extension).
The new extension includes the custom npm capability.

Read more about [why we created a new extension and what's new](https://www.jfrog.com/confluence/display/JFROG/Artifactory+Azure+DevOps+Extension#ArtifactoryAzureDevOpsExtension-WeCreatedaNewExtension).
The documentation of the new extension can be found [here](https://www.jfrog.com/confluence/display/JFROG/JFrog+Azure+DevOps+Extension).
We'd appreciate your feedback about the capability and the new extension once you try it out.
Thanks!
